h2o-core/src/main/java/water/fvec/RollupStats.java

```diff
@@ -431,7 +437,9 @@ public void compute2() {
             rs._tsk.join();
           } else if(rs.isMutating()) // c) => throw IAE
             throw new IllegalArgumentException("Can not compute rollup stats while vec is being modified. (3)");
-        } else { // d) => compute the rollups
+        } else { // rs == null d) => compute the rollups
+          final Vec vec = DKV.getGet(_vecKey);
+          if (vec == null) break; // Vec was deleted since we last checked
```

Comment:
I think I got it. 

The thing is, if the data is not there you do get an NPE but you already've inserted the computing RS which will remain in the K/V and leak (And this can still happen even with your check - the vec can be removed while the Roll task is in progress). 

The correct fix would then be to remove the computing RS if the Roll task fails.